src/parser/Statement.ts

```diff
@@ -1113,7 +1115,13 @@ export class NamespaceStatement extends Statement implements TypedefProvider {
     }
 
     public getName(parseMode: ParseMode) {
-        return this.nameExpression.getName(parseMode);
+        const fullyQualifiedNames = [
+            ...this.parentNameExpressions,
+            this.nameExpression
+        ].map(expression => expression.getName(parseMode));
+        const sep = parseMode === ParseMode.BrighterScript ? '.' : '_';
+
+        return fullyQualifiedNames.join(sep);
```

Comment:
Refactor this to leverage `.findAncestor()`.  Keep in mind, `.parent` is not set until the `program.validate()` cycle, but that's acceptable since many other BrighterScript AST features also rely on `.parent` as well.

Something like this:

```typescript
const parentNamespaces: NamespaceStatement[] = [this];
let ancestor = this;
//build a list of namespace ancestors
while (ancestor) {
    ancestor = this.findAncestor<NamespaceStatement>(isNamespaceStatement);
    if(ancestor){
        parentNamespaces.push(ancestor);
    }
}
//somehow build the fully qualified name now
```